### PR TITLE
Adding basic joystick support

### PR DESCRIPTION
**Pull request by maxfish**

### CODE REVIEW

File: examples/input/main.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	app := a.InitApp(640, 480, true, "Input")
+	defer a.TerminateApp()
+
+	joy := &input.JoystickController{}
+	joy.Open(0)
+
+	app.MainLoop(func(speed float64) {
+		// NOP
+	}, func() {
+		joy.Update()
```
[the remark this answers is not in the document]
Comment (maxfish) [reply examples/input/main.go]: yep... it should be :)

File: pkg/input/joystick_controller.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+func (c *JoystickController) Update() {
+	if !c.connected {
+		return
+	}
+
+	// Buttons
+	c.rawButtons = glfw.GetJoystickButtons(c.deviceIndex)
+	if c.buttonsDown == nil {
+		// builds the slices
+		c.buttonsDown = make([]bool, len(c.rawButtons))
```
[the remark this answers is not in the document]
Comment (maxfish) [reply pkg/input/joystick_controller.go]: If the joystick is disconnected, a different device (with another amount of buttons) might be plugged in and get the same deviceIndex.

File: pkg/input/joystick_controller.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		c.buttonsPressed = make([]bool, len(c.rawButtons))
+		c.buttonsReleased = make([]bool, len(c.rawButtons))
+	}
+
+	for i := 0; i < len(c.rawButtons); i++ {
+		isDown := c.rawButtons[i] > 0
+		c.buttonsPressed[i] = false
```
[the remark this answers is not in the document]
Comment (maxfish) [reply pkg/input/joystick_controller.go]: I know there would have been a better way, I was just too lazy for that. I'll fix it

File: pkg/input/joystick_controller.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			}
+		}
+		c.buttonsDown[i] = isDown
+	}
+
+	// Axes
+	c.axes = glfw.GetJoystickAxes(c.deviceIndex)
+}
+
+func (c *JoystickController) GetAxisValue(axisIndex int) (float32) {
```
[the remark this answers is not in the document]
Comment (maxfish) [reply pkg/input/joystick_controller.go]: That's the mapping of the joystick which is dependant on the model. GLFW (C version) is adding that now so I guess we could add a couple of mappings for XBox360 and PS4 at least

[the remark this answers is not in the document]
Comment (maxfish) [reply pkg/input/joystick_controller.go]: I'll introduce the type and the mapping in another pull request

File: pkg/input/joystick_controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (keymone) [reply pkg/input/joystick_controller.go]: right, but we can re-create these in .Open(), technically it's the only place where we really need to do it

[the remark this answers is not in the document]
Comment (maxfish) [reply pkg/input/joystick_controller.go]: It makes sense, I'm cleaning it up

**Pull request merged**